src/server/components/dashboard/controllers/controllers.ts

```diff
@@ -314,3 +316,24 @@ export function helpPageController(req: Request, res: Response): void {
     backUrl: req.session.currentUrl ?? "/dashboard/lists",
   });
 }
+
+export function userDeleteGetController(req: Request, res: Response) {
+  const { userEmail } = req.params;
+  res.render("dashboard/users-delete-confirm", {
+    ...DEFAULT_VIEW_PROPS,
+    userEmail,
+    req,
+  });
+}
+
+export async function userDeletePostController(req: Request, res: Response) {
+  const { userEmail } = req.params;
+  const userData = await findUserByEmail(userEmail); // TODO remove after testing
+  const userIsSuperAdmin = await isAdministrator(userEmail);
```

Comment:
We're checking if `userIsSuperAdmin` but not actually doing anything with it. 

I think we should be checking if the currently authenticated user is super admin though. i.e. only super admins can delete other users. 

is `findUserByEmail` just used to check if that user is safe to delete? should we just use `checkUserExists` for that? 

autheniticatedUser also has `emailAddress` as a property, so we don't have to do `req.user.userData.email` - `just req.user.emailAddress`.

```
const { userEmail } = req.params;

if(req.user?.isAdministrator) {
  const authenticatedUser = req.user!
  await deleteUserByEmail(userEmail, authenticatedUser.emailAddress)
}
```

Reply:
`userIsSuperAdmin` is only used to display different text in the flash message but I wasn't aware this information could be grabbed from the `req` object. Well I was aware I'd just forgotten 😅

Reply:
`findUserByEmail ` should be deleted. I was just using it to add the user id to the jsonData to check the correct details were being removed from the DB

Reply:
In terms of adding an if statement to check if the user is an admin `req.user?.isAdministrator` my assumption was that since the edit button is only visible to super admins that regular admins wouldn't be able to delete a user.

But they could change url so it's good to put the if statement there as a failsafe 🤷